### PR TITLE
Documentation 3.4/example generation improvements

### PR DESCRIPTION
**Pull request by Simran-B**
### Scope & Purpose

- choosable storage engine (or agnostic)
  -> required for hot backup documentation examples (RocksDB required)
- logCurlRequestPlain() / logPlainResponse() for better formatting (e.g. batch requests) of @EXAMPLE_ARANGOSH_RUN in DocuBlocks

Backport of #9498 (#9478, #9489 and some more fixes) and #9509 to 3.4

Preview: https://deploy-preview-105--zealous-morse-14392b.netlify.com/docs/3.4/http/batch-request.html#executes-a-batch-request

### CODE REVIEW

File: Documentation/Examples/WalPropertiesGet_mmfiles.generated [hunk 1]
```diff
@@ -0,0 +1,3 @@
+arangosh&gt; <span class="hljs-built_in">require</span>(<span class="hljs-string">"internal"</span>).wal.properties();
+{
```
Comment (dothebart) [inline Documentation/Examples/WalPropertiesGet_mmfiles.generated]: hm, this doesn't look good - is this really generated with mmfiles?

**Comment by Simran-B** [comment]
@dothebart Looks alright to me in the generated examples over in the my docs test PR:

https://github.com/arangodb/docs/blob/514a516d36392a89e37add8b8e15162586d93fd3/3.4/generated/Examples/WalPropertiesGet_mmfiles.generated
https://github.com/arangodb/docs/blob/514a516d36392a89e37add8b8e15162586d93fd3/3.4/generated/Examples/RestWalPropertiesGet_mmfiles.generated

It's probably just the file in this PR which was generated with RocksDB. None of the .generated files here should matter to the documentation system.

**Comment by dothebart** [comment]
would be good to have the proper files here anyway.

**Comment by Simran-B** [comment]
Regenerated examples:
https://github.com/arangodb/arangodb/pull/9557/commits/4b6ab838987306a75306d7fff13ac0e26f3e48ec#diff-c1aa492cbc71a57d7add7f1fb0de8944

**Pull request merged**